Automattic/vip-go-mu-plugins
# Exclude private taxonomies via filter

## Description
We are excluding some taxonomies in our fork of [EP](https://github.com/Automattic/ElasticPress/blob/0f8e520f2cdda2bea6450cf04e2bffcc71446116/includes/classes/Indexable/Post/Post.php#L1014).

The change comes from this [requirement](https://github.com/Automattic/ElasticPress/pull/13).

In one of the previous EP releases new filter `ep_post_tax_excluded_wp_query_root_check` was introduced that can be leveraged to do the same thing without diverging from upstream.

This change starts to use the filter. In follow-up change, we can remove our custom change inside EP fork.
Follow-up change: https://github.com/Automattic/ElasticPress/pull/133

## Changelog Description

### Plugin Updated: Enterprise Search

The first step towards moving excluded protected taxonomies from post queries. 

After this and follow-up change will be done, we will be able to enable the ability to change the list of excluded taxonomies instead of the current hard-coded solution.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

This is a bit tough. This is what I have done:
1) In EP - https://github.com/Automattic/ElasticPress/blob/0f8e520f2cdda2bea6450cf04e2bffcc71446116/includes/classes/Indexable/Post/Post.php#L1014 comment out lines 1015 - 1017 with our custom code.
2) Inside the `if` adding the taxanomy to `$args['tax_query']` add a debug line: `var_dump(['including taxanomy', $tag_slug])`
3) Add following code somewhere (can be a `mu-plugin`):
```
		add_filter( 'posts_pre_query', function($posts, $query) {
			$query->query_vars['fields'] = '123';
		}, 10, 2);
		add_action('init', function() {
			register_taxonomy( 'fields', ['post'] );
		});
```
4) Search on site -> you should see `fields` being included in the query in post_filters and the debug line shown
5) Apply the change and search again. `fields` is no longer included.

![image](https://user-images.githubusercontent.com/19240162/141298057-ebc9d431-9696-4fe3-a63c-43b0e8271367.png)
